ui/app/pages/swaps/swaps.util.js

```diff
@@ -265,17 +266,35 @@ export async function fetchTokenBalance (address, userAddress) {
   return usersToken
 }
 
-export function getRenderableGasFeesForQuote (tradeGas, approveGas, gasPrice, currentCurrency, conversionRate) {
+export function getRenderableNetworkFeesForQuote (
+  tradeGas,
+  approveGas,
+  gasPrice,
+  currentCurrency,
+  conversionRate,
+  tradeValue,
+  sourceSymbol,
+  sourceAmount,
+) {
   const totalGasLimitForCalculation = (new BigNumber(tradeGas || '0x0', 16)).plus(approveGas || '0x0', 16).toString(16)
   const gasTotalInWeiHex = calcGasTotal(totalGasLimitForCalculation, gasPrice)
 
+  const totalWeiCost = new BigNumber(gasTotalInWeiHex, 16)
+    .plus(tradeValue, 16)
+    .minus(sourceSymbol === 'ETH' ? sourceAmount : 0, 10)
+    .toString(16)
+
+  const nonGasFee = new BigNumber(totalWeiCost, 16)
+    .minus(gasTotalInWeiHex, 16)
+    .toString(16)
```

Comment:
Nit: we're now calculating this value twice - it might be simpler to calculate this once, first, and add it to the gas total.

```suggestion
  const nonGasFee = new BigNumber(tradeValue, 16)
    .minus(sourceSymbol === 'ETH' ? sourceAmount : 0, 10)
    .toString(16)

  const totalWeiCost = new BigNumber(gasTotalInWeiHex, 16)
    .plus(nonGasFee, 16)
    .toString(16)
```